File: solutions/27_Remove_Element.java

class Solution {
    public int removeElement(int[] nums, int val) {
        int counter = 0;

        for (int num : nums) {
            // Copy element to index of final array only if it is not equal to val
            if (num != val) {
                nums[counter] = num;
                counter++;
            }
        }
        return counter;
    }
}

File: solutions/27_Remove_Element.c


int removeElement(int* nums, int numsSize, int val) {
    // Keep track of index of the updated array
    int counter = 0;

    for (int i = 0; i < numsSize; i++) {
        // Copy element to index of final array only if it is not equal to val
        if (nums[i] != val) {
            nums[counter] = nums[i];
            counter++;
        }
    }
    return counter;
}